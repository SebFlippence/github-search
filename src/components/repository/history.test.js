import React from 'react';
import { render } from '@testing-library/react';

import RepositoryHistory from './history';

test('renders correctly when there is no history', () => {
  const { baseElement } = render(
    <RepositoryHistory commitContributionsByRepository={[]} />
  );
  expect(baseElement).toMatchSnapshot();
});

test('renders history correctly', () => {
  const { baseElement } = render(
    <RepositoryHistory
      commitContributionsByRepository={[
        {
          contributions: {
            nodes: [
              {
                repository: {
                  id: 'my-repo',
                  url: 'https://example.com',
                  name: 'My Repo',
                  ref: {
                    target: {
                      history: {
                        nodes: [
                          {
                            id: 'commit1',
                            authoredDate: '01/01/2020',
                            message: 'A commit message'
                          },
                          {
                            id: 'commit2',
                            authoredDate: '02/01/2020',
                            message: 'A second commit message'
                          }
                        ]
                      }
                    }
                  }
                }
              }
            ]
          }
        }
      ]}
    />
  );

  expect(baseElement).toMatchSnapshot();
});
