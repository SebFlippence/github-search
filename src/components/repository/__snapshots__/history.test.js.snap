// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is no history 1`] = `
<body>
  <div>
    <p
      class="font-bold"
    >
      No recent commits found
    </p>
  </div>
</body>
`;

exports[`renders history correctly 1`] = `
<body>
  <div>
    <h2
      class="block font-bold text-l pb-2 py-4"
    >
      Latest commits
    </h2>
    <ul>
      <li>
        <a
          class="underline font-bold"
          href="https://example.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          My Repo
        </a>
        <ol
          class="list-inside list-decimal"
        >
          <li
            class="truncate"
          >
            <span
              title="01/01/2020"
            >
              A commit message
            </span>
          </li>
          <li
            class="truncate"
          >
            <span
              title="02/01/2020"
            >
              A second commit message
            </span>
          </li>
        </ol>
      </li>
    </ul>
  </div>
</body>
`;
